### PR TITLE
Bump to 3.21.9.

### PR DESCRIPTION
**Pull request by jezdez**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This undoes the changes in https://github.com/conda-forge/conda-build-feedstock/pull/176 since conda-build 3.21.9 has been released now.

I also changed the URL similar how it's done in the conda feedstock, directly downloading from the git tag. We don't upload build artifacts anymore during release when we moved conda-build from Azure and GitHub Actions recently.

### CODE REVIEW

**Comment by jezdez** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
   version: {{ version }}
 
 source:
-  url: https://github.com/conda/conda-build/releases/download/{{ version }}/conda-build-{{ version }}.tar.gz
-  sha256: 0031aff4a34e66340f90428051d480c6c456c89e267f78cf5987126bee68ca5b
+  url: https://github.com/conda/{{ name }}/archive/{{ version }}.tar.gz
+  sha256: 19a5568e09ee476c7e9e68e00583f8ff1e4ca16b709e5078552156194ecd14ee
   patches:
     # backport conda/conda-build#4287
     - patches/0001-Remove-rpaths-that-occur-multiple-times.patch
```
Comment (dbast) [inline recipe/meta.yaml:13]: Does not apply and can be removed according to https://github.com/conda/conda-build/pull/4287
```suggestion
```

**Comment by jezdez** [comment]
Thanks @dbast!

File: recipe/meta.yaml [hunk 2]
```diff
@@ -137,3 +132,5 @@ extra:
     - xhochy
     - cjmartian
     - dbast
+    - jezdez
+    - chenglee
```
Comment (isuruf) [inline recipe/meta.yaml:136]: Shouldn't this be @chenghlee? (Note the missing 'h')

Comment (jezdez) [reply recipe/meta.yaml:136]: 🤦 Thanks both :)

**Pull request merged**